sdks/java/core/src/main/java/org/apache/beam/sdk/io/TextSource.java

```diff
@@ -171,21 +194,42 @@ protected void startReading(ReadableByteChannel channel) throws IOException {
           } else {
             startOfNextRecord = bufferPosn = (int) requiredPosition;
           }
+          skipHeader(skipHeaderLines, true);
         } else {
-          ((SeekableByteChannel) channel).position(requiredPosition);
-          startOfNextRecord = requiredPosition;
+          skipHeader(skipHeaderLines, false);
+          if (requiredPosition > startOfNextRecord) {
```

Comment:
I am wondering what will happen if multiple readers having requiredPosition <= startOfNextRecord here.
In this case, more than readers are assigned to a start position that fall in the header lines (assuming there are multiple long header lines). Not sure if the current code is handling this.

Reply:
good point